clang/lib/Sema/TreeTransform.h

```diff
@@ -13714,6 +13714,12 @@ TreeTransform<Derived>::TransformLambdaExpr(LambdaExpr *E) {
 
     // Capturing 'this' is trivial.
     if (C->capturesThis()) {
+      // We need ThisType here.
+      Sema::CXXThisScopeRAII ThisScope(
+          getSema(),
+          dyn_cast_or_null<CXXRecordDecl>(
```

Comment:
```suggestion
          dyn_cast_if_present<CXXRecordDecl>(
```
`cast_or_null` and friends are deprecated.